lib/middleman-simple-thumbnailer/extension.rb

```diff
@@ -37,12 +37,35 @@ def after_build(builder)
       def image_tag(path, options={})
         resize_to = options.delete(:resize_to)
         return super(path, options) unless resize_to
+        ext = app.extensions[:middleman_simple_thumbnailer]
 
-        image = MiddlemanSimpleThumbnailer::Image.new(path, resize_to, app)
         if app.development?
-          super("data:#{image.mime_type};base64,#{image.base64_data}", options)
+
+          # For better performance on local development
+          # (high bandwidth)
+          # simply change width in <img>-tag
+          
+          original_width  = options.delete(:width)
+          resized_width   = resize_to.split('x').first.to_i
+          original_height = options.delete(:height)
+          resized_height  = resize_to.split('x').last.to_i
+          
+          if !original_width.nil? && original_width <= resized_width
+            if original_height <= resized_height
+              options[:width] = original_width
```

Comment:
My use case is rather simple. I scale all images in certain folders down to a width of 850px. I then create an `<a href="image.jpg"><img src="image.850x2000.jpg"></a>` tag. Since I use the bootstrap class "image-responsive" for the image, I don't care much about the width- and height -tag.

So this could be accomplished with a script-shell `find | convert`-style solution. But I wanted to automate this at build-time and keep the scaled down-versions out of my git-repository.

After my crude change to the code, it behaves exactly the way I want it. (Having hundreds of megabytes included into blog-code that will be processed with `articles.each do` slowed things down and running imagemagick with every build drained my battery.)

Sidenote: I was thinking about using middleman-automatic-clown-car but that does not seem to be compatible with middleman 4 (at first glance).